dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/helpers/get-utilisation-report-reconciliation-details.test.ts

```diff
@@ -87,5 +100,55 @@ describe('get-utilisation-report-reconciliation-details-by-id.controller helpers
         keyingSheet: [],
       });
     });
+
+    it('filters the fee record payment groups by the facility id when the facility id is a string', async () => {
+      // Arrange
+      const uploadedReport = UtilisationReportEntityMockBuilder.forStatus('PENDING_RECONCILIATION')
+        .withId(reportId)
+        .withBankId(bankId)
+        .withDateUploaded(new Date())
+        .withFeeRecords([])
+        .build();
+
+      filterFeeRecordSpy.mockReturnValue([]);
+
+      const bankName = 'Test bank';
+      when(getBankNameById).calledWith(bankId).mockResolvedValue(bankName);
+      when(getKeyingSheetForReportId).calledWith(reportId, []).mockResolvedValue([]);
+      when(getFeeRecordPaymentEntityGroups).calledWith([]).mockReturnValue([]);
+      when(mapToFeeRecordPaymentGroups).calledWith([]).mockResolvedValue([]);
+
+      const facilityIdFilter = 'some filter';
+
+      // Act
+      await getUtilisationReportReconciliationDetails(uploadedReport, facilityIdFilter);
+
+      // Assert
+      expect(filterFeeRecordSpy).toHaveBeenCalledWith([], facilityIdFilter);
+    });
+
+    it('does not filter the fee record payment groups by the facility id when the facility id is undefined', async () => {
+      // Arrange
+      const uploadedReport = UtilisationReportEntityMockBuilder.forStatus('PENDING_RECONCILIATION')
+        .withId(reportId)
+        .withBankId(bankId)
+        .withDateUploaded(new Date())
+        .withFeeRecords([])
+        .build();
+
+      const bankName = 'Test bank';
+      when(getBankNameById).calledWith(bankId).mockResolvedValue(bankName);
+      when(getKeyingSheetForReportId).calledWith(reportId, []).mockResolvedValue([]);
+      when(getFeeRecordPaymentEntityGroups).calledWith([]).mockReturnValue([]);
+      when(mapToFeeRecordPaymentGroups).calledWith([]).mockResolvedValue([]);
```

Comment:
I would prefer not to. Each individual test may want its own set up with respect to how these functions are called, and abstracting away that detail through a test helper would make this harder for future developers to modify and maintain

Reply:
Seems to be the contrary to me - each test currently has the same stubs; If a future developer needs to modify these, it currently has to be modified in multiple places and this may not be obvious.

If they're in a single DRY function e.g at the top of the test, it's immediately obvious there's some repeated stubs and they can be changed in one place.

Reply:
Moved.